File: packages/windmill/src/tasks/import_voters_delegation.rs
// SPDX-FileCopyrightText: 2025 Sequent Tech <legal@sequentech.io>
//
// SPDX-License-Identifier: AGPL-3.0-only

use crate::types::error::{Error, Result};
use celery::error::TaskError;
use sequent_core::types::hasura::core::TasksExecution;
use serde::{Deserialize, Serialize};
use tracing::{error, info, instrument};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct ImportVotersDelegationInput {
    pub tenant_id: String,
    pub document_id: String,
    pub election_event_id: String,
    pub sha256: Option<String>,
}
#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct ImportVotersDelegationOutput {
    pub task_execution: TasksExecution,
}

#[instrument(err)]
#[wrap_map_err::wrap_map_err(TaskError)]
#[celery::task(max_retries = 2)]
pub async fn import_voters_delegation_task(
    body: ImportVotersDelegationInput,
    task_execution: TasksExecution,
) -> Result<()> {
    Ok(())
}

File: packages/admin-portal/src/queries/ImportVotersDelegation.ts
// SPDX-FileCopyrightText: 2025 Sequent Tech <legal@sequentech.io>
//
// SPDX-License-Identifier: AGPL-3.0-only
import {gql} from "@apollo/client"

export const IMPORT_VOTERS_DELEGATION = gql`
    mutation ImportVotersDelegation(
        $tenantId: String!
        $electionEventId: String!
        $documentId: String!
    ) {
        import_voters_delegation(
            tenant_id: $tenantId
            election_event_id: $electionEventId
            document_id: $documentId
        ) {
            task_execution {
                id
                name
                execution_status
                created_at
                start_at
                end_at
                logs
                annotations
                labels
                executed_by_user
                tenant_id
                election_event_id
                type
            }
        }
    }
`
